### PR TITLE
[PDI-17667] Use of vulnerable component jetty 8.1.15, CVE-2017-9735, …

### PR DESCRIPTION
**Pull request by smmribeiro**
…CVE-2017-7657 CVE-2015-2080

[PPP-4183] Use of Vulnerable Component: Multiple Jetty Components [Listed in Description] (CVE-2017-7656 | CVE-2017-7657 | CVE-2017-7658 | CVE-2015-2080 | CVE-2017-9735 )

Centralized Jetty components adding a property for the Jetty version (upgrading it to 9.4.15.v20190215) and making the needed declarations at the dependency management level.

Dependent PRs:
- pentaho/big-data-plugin#1662
- pentaho/pdi-monitoring-plugin#94
- pentaho/pdi-sdk-plugins#74
- pentaho/pentaho-kettle#6398
- pentaho/pentaho-platform#4398
- pentaho/pentaho-platform-plugin-reporting#707
- pentaho/pentaho-reporting#1255
- pentaho/pentaho-yarn-kettle-plugin#372
- webdetails/cde#928

@pentaho-lmartins @pentaho/tatooine 
@graimundo @nantunes

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -266,6 +267,101 @@
...
+        <artifactId>jetty-io</artifactId>
+        <version>${jetty.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.eclipse.jetty</groupId>
+        <artifactId>jetty-jaas</artifactId>
+        <version>${jetty.version}</version>
+        <exclusions>
+          <exclusion>
+            <groupId>*</groupId>
```
Comment (graimundo) [inline pom.xml]: Why the exclusions?

File: pom.xml [hunk 2]
```diff
@@ -266,6 +267,101 @@
...
+        <artifactId>jetty-util</artifactId>
+        <version>${jetty.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.eclipse.jetty</groupId>
+        <artifactId>jetty-util-ajax</artifactId>
+        <version>${jetty.version}</version>
+        <exclusions>
+          <exclusion>
+            <artifactId>*</artifactId>
```
Comment (graimundo) [inline pom.xml]: Why the exclusions?

File: pom.xml [hunk 3]
```diff
@@ -266,6 +267,101 @@
...
+          </exclusion>
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>org.eclipse.jetty</groupId>
+        <artifactId>jetty-webapp</artifactId>
+        <version>${jetty.version}</version>
+        <exclusions>
+          <exclusion>
+            <groupId>*</groupId>
```
Comment (graimundo) [inline pom.xml]: Why the exclusions?

File: pom.xml [hunk 4]
```diff
@@ -266,6 +267,101 @@
...
+        <version>${jetty.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.eclipse.jetty</groupId>
+        <artifactId>test-jetty-servlet</artifactId>
+        <version>${jetty.version}</version>
+        <scope>test</scope>
+        <exclusions>
+          <exclusion>
+            <groupId>*</groupId>
```
Comment (graimundo) [inline pom.xml]: Why the exclusions?

File: pom.xml (hunk 3 above)
Comment (smmribeiro) [reply pom.xml]: As the only usage of these artifacts had these exclusions, I've simply moved the exclusions to the parent pom.
Is it preferable that I move the exclusions to the poms where the dependency is used?

File: pom.xml (hunk 2 above)
Comment (smmribeiro) [reply pom.xml]: As the only usage of these artifacts had these exclusions, I've simply moved the exclusions to the parent pom.
Is it preferable that I move the exclusions to the poms where the dependency is used?

File: pom.xml (hunk 1 above)
Comment (smmribeiro) [reply pom.xml]: As the only usage of these artifacts had these exclusions, I've simply moved the exclusions to the parent pom.
Is it preferable that I move the exclusions to the poms where the dependency is used?

File: pom.xml (hunk 4 above)
Comment (smmribeiro) [reply pom.xml]: @graimundo , as the only usage of these artifacts had these exclusions, I've simply moved the exclusions to the parent pom.
Is it preferable that I move the exclusions to the poms where the dependency is used?

File: pom.xml (hunk 1 above)
Comment (graimundo) [reply pom.xml]: > Is it preferable that I move the exclusions to the poms where the dependency is used?
Yes. Unless there is a good reason to exclude at the top level of all projects the transitives, the exclusions should be at the "leaf" projects/modules. The same applies to the remaining exclusions in this PR.

Comment (smmribeiro) [reply pom.xml]: Ok, will do that!
I'll resolve the other conversations for they're the same as this one.

Comment (smmribeiro) [reply pom.xml]: Pushed new code for this PR: resolving this conversation.
New conversations may be created when reviewing the new code - I hope not :-)

**Comment by smmribeiro** [comment]
Closing PR: to simplify Karaf's upgrade a different approach was defined!

**Pull request closed without merging**